sdks/java/core/src/main/java/org/apache/beam/sdk/options/GcsOptions.java

```diff
@@ -121,10 +120,7 @@
     public ExecutorService create(PipelineOptions options) {
       ThreadFactoryBuilder threadFactoryBuilder = new ThreadFactoryBuilder();
       threadFactoryBuilder.setThreadFactory(MoreExecutors.platformThreadFactory());
-      if (!AppEngineEnvironment.IS_APP_ENGINE) {
-        // AppEngine doesn't allow modification of threads to be daemon threads.
-        threadFactoryBuilder.setDaemon(true);
-      }
+      threadFactoryBuilder.setDaemon(true);
```

Comment:
Do we really want to make these daemon threads all the time?

Previously on the Dataflow worker and SDK harness, these were non daemon threads.

The GcsOptions ExecutorService is consumed and used in several places.